prometheus_client/metrics.py

```diff
@@ -1,17 +1,23 @@
 from threading import Lock
 import time
 import types
+from typing import (
+    Any, Callable, Dict, Optional, Sequence, Tuple, Type, TypeVar,
+)
 
 from . import values  # retain this import style for testability
 from .context_managers import ExceptionCounter, InprogressTracker, Timer
 from .metrics_core import (
     Metric, METRIC_LABEL_NAME_RE, METRIC_NAME_RE,
     RESERVED_METRIC_LABEL_NAME_RE,
 )
-from .registry import REGISTRY
+from .registry import CollectorRegistry, REGISTRY
 from .samples import Exemplar
 from .utils import floatToGoString, INF
 
+T = TypeVar('T', bound='MetricWrapperBase')
```

Comment:
Rather than using `T` could we just use `"MetricWrapperBase"` to make everything clearer later on when reading the functions?

Reply:
That wouldn't work the same way.

This type is used as a generic, it's basically bound to the class and ensures that type returned is of the same type as the mothod's class. Probably I might not even need the bound parameter here.

Reply:
Ahh, yeah that makes sense. Thanks!